src/components/comment/comment.ts

```diff
@@ -0,0 +1,41 @@
+export const initCommentsWithRepliesEnabled = () => {
+  const commentsWithRepliesEnabled = document.querySelectorAll(
+    '.js-comment-with-replies-enabled'
+  );
+
+  for (const comment of [...commentsWithRepliesEnabled]) {
+    const replyButton = comment.querySelector('.js-comment-reply-button');
+    const replyForm = comment.querySelector('.js-comment-reply-form');
+    // The cancel button is inside the Comment Form, so we need to reach inside
+    // that component to grab the button. This feels a little awkward since
+    // we're directly manipulating a child component, but this is a lot simpler
+    // than other potential solutions. Other options included:
+    // - Having the Comment Form emit a `cancel` event
+    // - Having the Comment Form have a block that you could pass a button into
+    const cancelButton = comment.querySelector('.js-cancel-reply');
+
+    replyButton?.addEventListener('click', () => {
+      comment.classList.add('is-replying');
+      replyButton.setAttribute('hidden', '');
+      replyForm?.removeAttribute('hidden');
+      const firstInput = replyForm?.querySelector(
+        'textarea, input'
+      ) as HTMLElement | null;
+
+      // Without this timeout, VoiceOver does not properly focus the first input
+      // (though it works outside of VoiceOver).
+      // I went back and forth on whether or not to force this behavior in
+      // VoiceOver, since it will skip them past context around the comment
+      // (e.g. you may use simple HTML or markdown).
+      // I ended up adding this timeout to force the focus so that we provide
+      // an equivalent experience for all users.
+      setTimeout(() => firstInput?.focus(), 0);
+    });
+
+    cancelButton?.addEventListener('click', () => {
+      comment.classList.remove('is-replying');
+      replyButton?.removeAttribute('hidden');
+      replyForm?.setAttribute('hidden', '');
+    });
+  }
```

Comment:
@Paul-Hebert Should we return a cleanup function to remove the event listeners, like we've done with the other JS files? https://github.com/cloudfour/cloudfour.com-patterns/blob/v-next/src/components/sky-nav/sky-nav.ts#L111-L119

Reply:
This is a good question. I just went down a rabbit hole of SO answers and I'm still not sure:

- https://stackoverflow.com/questions/12528049/if-a-dom-element-is-removed-are-its-listeners-also-removed-from-memory
- https://stackoverflow.com/questions/18986520/does-removing-an-element-also-remove-its-event-listeners
- https://stackoverflow.com/questions/53256662/react-why-should-i-remove-event-listeners

It seems like it would be safest to return a cleanup function. I'll add one.

Reply:
I see no harm in providing an API to remove event listeners. If the component consumer doesn't need it, no harm is done. If it is needed, though, it will be appreciated.

Reply:
@Paul-Hebert Once you add it we'll also want to use the cleanup function within storybook, as a cleanup function for the useEffect hook. I'm not sure how storybook's useEffect compares to React's, but React's useEffect hook reruns after any prop change, so if you change props, right now it probably keeps reattaching more and more event listeners